{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1b9a5fac",
   "metadata": {},
   "source": [
    "* Read in and transform parquets into single files and load them\n",
    "* Data from openTargets - https://platform.opentargets.org/downloads\n",
    "\n",
    "\n",
    "ALSO - ICD10 occurences, using table from UKBB and nadav Rap -\n",
    "* `ICD10_counts.csv`\n",
    "```\n",
    "a table with counts (if you want frequencies, divide by 502616) of ICD10 codes. I created two columns, one that counts occurrences in the field 41202, the other is any of 41202, 40006, 41204\n",
    "```\n",
    "* Getting the ICD codes from diseases file (json or parquet):\n",
    "* https://community.opentargets.org/t/how-to-get-icd-codes-for-diseases/900/3\n",
    "```\n",
    "import pandas as pd\n",
    "path = 'some path to files'\n",
    "ref_filter = r\"^ICD[9,10].+\"\n",
    "raw_df: pd.DataFrame = pd.read_parquet(path, columns=['id', 'dbXRefs'])\n",
    "id_and_xref_df: pd.DataFrame = raw_df.explode('dbXRefs').astype('string')\n",
    "id_and_icd_df: pd.DataFrame = id_and_xref_df[id_and_xref_df[\"dbXRefs\"].str.match(\n",
    "    ref_filter)]\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "594d8ee7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os,sys\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2157a2c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install pyarrow fastparquet pandas -U"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0d2fe13f",
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_data_directory = \".\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "02bae3fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['./go',\n",
       " './evidence',\n",
       " './diseaseToPhenotype',\n",
       " './targets',\n",
       " './associationByDatasourceDirect',\n",
       " './associationByDatatypeDirect',\n",
       " './fda/significantAdverseDrugReactions',\n",
       " './fda',\n",
       " './mechanismOfAction',\n",
       " './indication',\n",
       " './associationByOverallDirect',\n",
       " './associationByOverallIndirect',\n",
       " './diseases',\n",
       " './drugWarnings',\n",
       " './mousePhenotypes',\n",
       " './molecule',\n",
       " './hpo',\n",
       " './interaction']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dirs = [x[0] for x in os.walk(raw_data_directory,topdown=False) if (\"/proc\" not in x)]\n",
    "bad_res = ['.','./.ipynb_checkpoints','./proc']\n",
    "dirs = [x for x in dirs if not any(x == c for c in bad_res)]\n",
    "\n",
    "\n",
    "### does excluding these mean no double counting of evidence????\n",
    "sub_ev = [ './evidence/sourceId=progeny',\n",
    " './evidence/sourceId=chembl',\n",
    " './evidence/sourceId=slapenrich',\n",
    " './evidence/sourceId=cancer_gene_census',\n",
    " './evidence/sourceId=intogen',\n",
    " './evidence/sourceId=expression_atlas',\n",
    " './evidence/sourceId=europepmc',\n",
    " './evidence/sourceId=eva_somatic',\n",
    " './evidence/sourceId=genomics_england',\n",
    " './evidence/sourceId=crispr',\n",
    " './evidence/sourceId=sysbio',\n",
    " './evidence/sourceId=impc',\n",
    " './evidence/sourceId=reactome',\n",
    " './evidence/sourceId=gene_burden',\n",
    " './evidence/sourceId=gene2phenotype',\n",
    " './evidence/sourceId=cancer_biomarkers',\n",
    " './evidence/sourceId=ot_genetics_portal',\n",
    " './evidence/sourceId=orphanet',\n",
    " './evidence/sourceId=uniprot_literature',\n",
    " './evidence/sourceId=eva',\n",
    " './evidence/sourceId=clingen',\n",
    " './evidence/sourceId=uniprot_variants',\n",
    "#          './fda/significantAdverseDrugReactions', # unneeded, and included in /fda - ?\n",
    "          \n",
    "#           './evidence', # save time on rerun\n",
    "         ]\n",
    "dirs = [x for x in dirs if not any(x == c for c in sub_ev)]\n",
    "\n",
    "dirs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3ecbeccc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dirs = [\n",
    "# ## todo: rerun with evidences\n",
    "#         './go',\n",
    "        \n",
    "#  './evidence',\n",
    "#  './evidence/sourceId=chembl',\n",
    "#  './evidence/sourceId=cancer_gene_census',\n",
    "#  './evidence/sourceId=europepmc',\n",
    "#  './evidence/sourceId=crispr',\n",
    "#  './evidence/sourceId=cancer_biomarkers',\n",
    "#  './evidence/sourceId=clingen',\n",
    "#  './diseaseToPhenotype',\n",
    "#  './targets',\n",
    "#  './associationByDatasourceDirect',\n",
    "#  './fda',\n",
    "#  './fda/significantAdverseDrugReactions',\n",
    "#  './mechanismOfAction',\n",
    "#  './indication',\n",
    "#  './associationByOverallDirect',\n",
    "#  './associationByOverallIndirect',\n",
    "#  './diseases',\n",
    "#  './mousePhenotypes',\n",
    "#  './molecule',\n",
    "#  './hpo',\n",
    "#  './interaction'\n",
    "#        ]\n",
    "\n",
    "# dirs = [\n",
    "# #     './drugWarnings',\n",
    "#  './mousePhenotypes',\n",
    "#  './molecule',\n",
    "#  './hpo',\n",
    "#  './interaction']\n",
    "\n",
    "\n",
    "dirs = [\n",
    "#     './drugWarnings',\n",
    " './diseases',\n",
    "# './diseaseToPhenotype',\n",
    "# #      './associationByOverallDirect',\n",
    "#     \"./go\",\n",
    "#  './targets',\n",
    "#     './hpo',\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b632c40",
   "metadata": {},
   "source": [
    "\"Q600 is indeed Q60.0, and those is true for all other codes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a741e1aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f41202</th>\n",
       "      <th>All</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ICD10</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C457</th>\n",
       "      <td>37</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S229</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M628</th>\n",
       "      <td>76</td>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y733</th>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T294</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R401</th>\n",
       "      <td>15</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X126</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I050</th>\n",
       "      <td>58</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M6080</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D169</th>\n",
       "      <td>18</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11491 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       f41202  All\n",
       "ICD10             \n",
       "C457       37   52\n",
       "S229        2    2\n",
       "M628       76  184\n",
       "Y733        0    5\n",
       "T294        1    1\n",
       "...       ...  ...\n",
       "R401       15   36\n",
       "X126        0    1\n",
       "I050       58  130\n",
       "M6080       2    2\n",
       "D169       18   22\n",
       "\n",
       "[11491 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_icd_counts_ukbb = pd.read_csv(\"ICD10_counts.csv\")\n",
    "df_icd_counts_ukbb[\"ICD10\"] = df_icd_counts_ukbb[\"ICD10\"].str.strip()\n",
    "df_icd_counts_ukbb.set_index(\"ICD10\",inplace=True)\n",
    "df_icd_counts_ukbb#[\"ICD10\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93193a73",
   "metadata": {},
   "source": [
    "```\n",
    "\"Q600 is indeed Q60.0, and those is true for all other codes.\"\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "592c4e7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>dbXRefs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MONDO_0007097</td>\n",
       "      <td>ICD10CM:E85.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MONDO_0007216</td>\n",
       "      <td>ICD10CM:Q73.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               id        dbXRefs\n",
       "10  MONDO_0007097  ICD10CM:E85.1\n",
       "11  MONDO_0007216  ICD10CM:Q73.8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "id         10110\n",
      "dbXRefs     2949\n",
      "ICD10       1231\n",
      "ICD10_2     2949\n",
      "dtype: int64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/hl/bb0z87yd5pzfcp81jc8662kr0000gn/T/ipykernel_61553/1169054070.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  id_and_icd_df[\"dbXRefs\"] =id_and_icd_df[\"dbXRefs\"].str.split(\":\",expand=True)[1]\n",
      "/var/folders/hl/bb0z87yd5pzfcp81jc8662kr0000gn/T/ipykernel_61553/1169054070.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  id_and_icd_df[\"ICD10\"] =id_and_icd_df[\"dbXRefs\"].str.split(\".\",expand=True)[0].str.strip()\n",
      "/var/folders/hl/bb0z87yd5pzfcp81jc8662kr0000gn/T/ipykernel_61553/1169054070.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  id_and_icd_df[\"ICD10_2\"] =id_and_icd_df[\"dbXRefs\"].str.replace(r\".\",\"\",regex=False).str.strip()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>dbXRefs</th>\n",
       "      <th>ICD10</th>\n",
       "      <th>ICD10_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MONDO_0007097</td>\n",
       "      <td>E85.1</td>\n",
       "      <td>E85</td>\n",
       "      <td>E851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MONDO_0007216</td>\n",
       "      <td>Q73.8</td>\n",
       "      <td>Q73</td>\n",
       "      <td>Q738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MONDO_0007604</td>\n",
       "      <td>Q87.8</td>\n",
       "      <td>Q87</td>\n",
       "      <td>Q878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MONDO_0008291</td>\n",
       "      <td>Q82.8</td>\n",
       "      <td>Q82</td>\n",
       "      <td>Q828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MONDO_0008695</td>\n",
       "      <td>E78.6</td>\n",
       "      <td>E78</td>\n",
       "      <td>E786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23066</th>\n",
       "      <td>Orphanet_294939</td>\n",
       "      <td>Q69.1</td>\n",
       "      <td>Q69</td>\n",
       "      <td>Q691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23068</th>\n",
       "      <td>Orphanet_352734</td>\n",
       "      <td>E70.3</td>\n",
       "      <td>E70</td>\n",
       "      <td>E703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23069</th>\n",
       "      <td>Orphanet_721</td>\n",
       "      <td>D69.1</td>\n",
       "      <td>D69</td>\n",
       "      <td>D691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23070</th>\n",
       "      <td>Orphanet_93322</td>\n",
       "      <td>Q72.5</td>\n",
       "      <td>Q72</td>\n",
       "      <td>Q725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23071</th>\n",
       "      <td>Orphanet_93925</td>\n",
       "      <td>Q04.2</td>\n",
       "      <td>Q04</td>\n",
       "      <td>Q042</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12016 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id dbXRefs ICD10 ICD10_2\n",
       "10       MONDO_0007097   E85.1   E85    E851\n",
       "11       MONDO_0007216   Q73.8   Q73    Q738\n",
       "12       MONDO_0007604   Q87.8   Q87    Q878\n",
       "13       MONDO_0008291   Q82.8   Q82    Q828\n",
       "14       MONDO_0008695   E78.6   E78    E786\n",
       "...                ...     ...   ...     ...\n",
       "23066  Orphanet_294939   Q69.1   Q69    Q691\n",
       "23068  Orphanet_352734   E70.3   E70    E703\n",
       "23069     Orphanet_721   D69.1   D69    D691\n",
       "23070   Orphanet_93322   Q72.5   Q72    Q725\n",
       "23071   Orphanet_93925   Q04.2   Q04    Q042\n",
       "\n",
       "[12016 rows x 4 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "path = './diseases'\n",
    "ref_filter = r\"^ICD[10].+\" #r\"^ICD[9,10].+\"\n",
    "raw_df: pd.DataFrame = pd.read_parquet(path, columns=['id', 'dbXRefs'])\n",
    "id_and_xref_df: pd.DataFrame = raw_df.explode('dbXRefs').astype('string')\n",
    "id_and_icd_df: pd.DataFrame = id_and_xref_df[id_and_xref_df[\"dbXRefs\"].str.match(\n",
    "    ref_filter)]\n",
    "\n",
    "display(id_and_icd_df.head(2))\n",
    "# ## DROP CM/CMS cases\n",
    "# id_and_icd_df = id_and_icd_df.loc[~id_and_icd_df[\"dbXRefs\"].str.startswith(\"ICD10CM\")]\n",
    "\n",
    "# note : this'll also include \"ICD10CM:\" if not dropped before\n",
    "id_and_icd_df[\"dbXRefs\"] =id_and_icd_df[\"dbXRefs\"].str.split(\":\",expand=True)[1]\n",
    "## get ~\"3 digit\" codes only\n",
    "id_and_icd_df[\"ICD10\"] =id_and_icd_df[\"dbXRefs\"].str.split(\".\",expand=True)[0].str.strip()\n",
    "## alt col without dot and as long number\n",
    "id_and_icd_df[\"ICD10_2\"] =id_and_icd_df[\"dbXRefs\"].str.replace(r\".\",\"\",regex=False).str.strip()\n",
    "print(id_and_icd_df.nunique())\n",
    "id_and_icd_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0fc0d42e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>dbXRefs</th>\n",
       "      <th>ICD10</th>\n",
       "      <th>All_ICD_counts</th>\n",
       "      <th>f41202_ICD_counts</th>\n",
       "      <th>ICD10_all_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MONDO_0007097</td>\n",
       "      <td>E85.1</td>\n",
       "      <td>E85</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MONDO_0007216</td>\n",
       "      <td>Q73.8</td>\n",
       "      <td>Q73</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MONDO_0007604</td>\n",
       "      <td>Q87.8</td>\n",
       "      <td>Q87</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.005571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MONDO_0008291</td>\n",
       "      <td>Q82.8</td>\n",
       "      <td>Q82</td>\n",
       "      <td>86.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>0.017110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MONDO_0008695</td>\n",
       "      <td>E78.6</td>\n",
       "      <td>E78</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12011</th>\n",
       "      <td>Orphanet_294939</td>\n",
       "      <td>Q69.1</td>\n",
       "      <td>Q69</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12012</th>\n",
       "      <td>Orphanet_352734</td>\n",
       "      <td>E70.3</td>\n",
       "      <td>E70</td>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.002984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12013</th>\n",
       "      <td>Orphanet_721</td>\n",
       "      <td>D69.1</td>\n",
       "      <td>D69</td>\n",
       "      <td>14.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.002785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12014</th>\n",
       "      <td>Orphanet_93322</td>\n",
       "      <td>Q72.5</td>\n",
       "      <td>Q72</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12015</th>\n",
       "      <td>Orphanet_93925</td>\n",
       "      <td>Q04.2</td>\n",
       "      <td>Q04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12016 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id dbXRefs ICD10  All_ICD_counts  f41202_ICD_counts  \\\n",
       "0        MONDO_0007097   E85.1   E85             2.0                1.0   \n",
       "1        MONDO_0007216   Q73.8   Q73             NaN                NaN   \n",
       "2        MONDO_0007604   Q87.8   Q87            28.0                5.0   \n",
       "3        MONDO_0008291   Q82.8   Q82            86.0               52.0   \n",
       "4        MONDO_0008695   E78.6   E78            27.0                0.0   \n",
       "...                ...     ...   ...             ...                ...   \n",
       "12011  Orphanet_294939   Q69.1   Q69             NaN                NaN   \n",
       "12012  Orphanet_352734   E70.3   E70            15.0                2.0   \n",
       "12013     Orphanet_721   D69.1   D69            14.0                5.0   \n",
       "12014   Orphanet_93322   Q72.5   Q72             NaN                NaN   \n",
       "12015   Orphanet_93925   Q04.2   Q04             NaN                NaN   \n",
       "\n",
       "       ICD10_all_freq  \n",
       "0            0.000398  \n",
       "1                 NaN  \n",
       "2            0.005571  \n",
       "3            0.017110  \n",
       "4            0.005372  \n",
       "...               ...  \n",
       "12011             NaN  \n",
       "12012        0.002984  \n",
       "12013        0.002785  \n",
       "12014             NaN  \n",
       "12015             NaN  \n",
       "\n",
       "[12016 rows x 6 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_and_icd_df = id_and_icd_df.merge(df_icd_counts_ukbb,on=[\"ICD10\"],how=\"left\")\n",
    "id_and_icd_df = id_and_icd_df.merge(df_icd_counts_ukbb,left_on=[\"ICD10_2\"],right_on=[\"ICD10\"],how=\"left\")\n",
    "id_and_icd_df[\"All_ICD_counts\"] = id_and_icd_df[\"All_x\"].fillna(id_and_icd_df[\"All_y\"])\n",
    "id_and_icd_df[\"f41202_ICD_counts\"] = id_and_icd_df[\"f41202_x\"].fillna(id_and_icd_df[\"f41202_y\"])\n",
    "# normalized %)\n",
    "id_and_icd_df = id_and_icd_df[['id', 'dbXRefs', 'ICD10','All_ICD_counts', 'f41202_ICD_counts']]\n",
    "\n",
    "id_and_icd_df[\"ICD10_all_freq\"] = 100*id_and_icd_df[\"All_ICD_counts\"]/502616\n",
    "id_and_icd_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ad45adee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>dbXRefs</th>\n",
       "      <th>ICD10</th>\n",
       "      <th>All_ICD_counts</th>\n",
       "      <th>f41202_ICD_counts</th>\n",
       "      <th>ICD10_all_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MONDO_0007216</td>\n",
       "      <td>Q73.8</td>\n",
       "      <td>Q73</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MONDO_0009061</td>\n",
       "      <td>E84</td>\n",
       "      <td>E84</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MONDO_0010080</td>\n",
       "      <td>G23.2</td>\n",
       "      <td>G23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MONDO_0008322</td>\n",
       "      <td>Q77.8</td>\n",
       "      <td>Q77</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>MONDO_0011348</td>\n",
       "      <td>Q69.0</td>\n",
       "      <td>Q69</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11040</th>\n",
       "      <td>Orphanet_797</td>\n",
       "      <td>D86</td>\n",
       "      <td>D86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11090</th>\n",
       "      <td>Orphanet_93976</td>\n",
       "      <td>Q16.0</td>\n",
       "      <td>Q16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11594</th>\n",
       "      <td>Orphanet_98602</td>\n",
       "      <td>H06</td>\n",
       "      <td>H06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11643</th>\n",
       "      <td>Orphanet_294944</td>\n",
       "      <td>M21</td>\n",
       "      <td>M21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11832</th>\n",
       "      <td>Orphanet_294947</td>\n",
       "      <td>M20</td>\n",
       "      <td>M20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>993 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id dbXRefs ICD10  All_ICD_counts  f41202_ICD_counts  \\\n",
       "1        MONDO_0007216   Q73.8   Q73             NaN                NaN   \n",
       "9        MONDO_0009061     E84   E84             NaN                NaN   \n",
       "14       MONDO_0010080   G23.2   G23             NaN                NaN   \n",
       "24       MONDO_0008322   Q77.8   Q77             NaN                NaN   \n",
       "33       MONDO_0011348   Q69.0   Q69             NaN                NaN   \n",
       "...                ...     ...   ...             ...                ...   \n",
       "11040     Orphanet_797     D86   D86             NaN                NaN   \n",
       "11090   Orphanet_93976   Q16.0   Q16             NaN                NaN   \n",
       "11594   Orphanet_98602     H06   H06             NaN                NaN   \n",
       "11643  Orphanet_294944     M21   M21             NaN                NaN   \n",
       "11832  Orphanet_294947     M20   M20             NaN                NaN   \n",
       "\n",
       "       ICD10_all_freq  \n",
       "1                 NaN  \n",
       "9                 NaN  \n",
       "14                NaN  \n",
       "24                NaN  \n",
       "33                NaN  \n",
       "...               ...  \n",
       "11040             NaN  \n",
       "11090             NaN  \n",
       "11594             NaN  \n",
       "11643             NaN  \n",
       "11832             NaN  \n",
       "\n",
       "[993 rows x 6 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_and_icd_df.loc[id_and_icd_df[\"All_ICD_counts\"].isna()].drop_duplicates(subset=[\"ICD10\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a000a02",
   "metadata": {},
   "source": [
    "#### Only ~ 75% coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4f20366a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "id                   8445\n",
      "dbXRefs              1518\n",
      "ICD10                 663\n",
      "All_ICD_counts        441\n",
      "f41202_ICD_counts     339\n",
      "ICD10_all_freq        441\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>dbXRefs</th>\n",
       "      <th>ICD10</th>\n",
       "      <th>All_ICD_counts</th>\n",
       "      <th>f41202_ICD_counts</th>\n",
       "      <th>ICD10_all_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MONDO_0007097</td>\n",
       "      <td>E85.1</td>\n",
       "      <td>E85</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MONDO_0007604</td>\n",
       "      <td>Q87.8</td>\n",
       "      <td>Q87</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.005571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MONDO_0008291</td>\n",
       "      <td>Q82.8</td>\n",
       "      <td>Q82</td>\n",
       "      <td>86.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>0.017110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MONDO_0008695</td>\n",
       "      <td>E78.6</td>\n",
       "      <td>E78</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MONDO_0008796</td>\n",
       "      <td>Q87.8</td>\n",
       "      <td>Q87</td>\n",
       "      <td>28.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.005571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12006</th>\n",
       "      <td>Orphanet_93457</td>\n",
       "      <td>Q73.1</td>\n",
       "      <td>Q73</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12007</th>\n",
       "      <td>Orphanet_95699</td>\n",
       "      <td>E25.0</td>\n",
       "      <td>E25</td>\n",
       "      <td>19.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.003780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12010</th>\n",
       "      <td>Orphanet_293925</td>\n",
       "      <td>Q87.5</td>\n",
       "      <td>Q87</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12012</th>\n",
       "      <td>Orphanet_352734</td>\n",
       "      <td>E70.3</td>\n",
       "      <td>E70</td>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.002984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12013</th>\n",
       "      <td>Orphanet_721</td>\n",
       "      <td>D69.1</td>\n",
       "      <td>D69</td>\n",
       "      <td>14.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.002785</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9419 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id dbXRefs ICD10  All_ICD_counts  f41202_ICD_counts  \\\n",
       "0        MONDO_0007097   E85.1   E85             2.0                1.0   \n",
       "2        MONDO_0007604   Q87.8   Q87            28.0                5.0   \n",
       "3        MONDO_0008291   Q82.8   Q82            86.0               52.0   \n",
       "4        MONDO_0008695   E78.6   E78            27.0                0.0   \n",
       "5        MONDO_0008796   Q87.8   Q87            28.0                5.0   \n",
       "...                ...     ...   ...             ...                ...   \n",
       "12006   Orphanet_93457   Q73.1   Q73             1.0                0.0   \n",
       "12007   Orphanet_95699   E25.0   E25            19.0                9.0   \n",
       "12010  Orphanet_293925   Q87.5   Q87             2.0                0.0   \n",
       "12012  Orphanet_352734   E70.3   E70            15.0                2.0   \n",
       "12013     Orphanet_721   D69.1   D69            14.0                5.0   \n",
       "\n",
       "       ICD10_all_freq  \n",
       "0            0.000398  \n",
       "2            0.005571  \n",
       "3            0.017110  \n",
       "4            0.005372  \n",
       "5            0.005571  \n",
       "...               ...  \n",
       "12006        0.000199  \n",
       "12007        0.003780  \n",
       "12010        0.000398  \n",
       "12012        0.002984  \n",
       "12013        0.002785  \n",
       "\n",
       "[9419 rows x 6 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_and_icd_df = id_and_icd_df.dropna(subset=[\"All_ICD_counts\",\"f41202_ICD_counts\"],how=\"all\",axis=0)\n",
    "print(id_and_icd_df.dropna(axis=0,thresh=5).nunique())\n",
    "id_and_icd_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9d63baff",
   "metadata": {},
   "outputs": [],
   "source": [
    "id_and_icd_df.to_csv(\"disease_icd10_counts_v1.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "292e79e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# id_and_icd_df.loc[id_and_icd_df[\"ICD10\"].isin(df_icd_counts_ukbb[\"ICD10\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "5d80e897",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Not in:\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'ICD10'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3628\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3629\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3630\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'ICD10'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/hl/bb0z87yd5pzfcp81jc8662kr0000gn/T/ipykernel_44425/1466061048.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Not in:\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mid_and_icd_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m~\u001b[0m\u001b[0mid_and_icd_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"ICD10\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_icd_counts_ukbb\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"ICD10\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdrop_duplicates\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msubset\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"ICD10\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3503\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3504\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3505\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3506\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3507\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3629\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3630\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3631\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3632\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3633\u001b[0m                 \u001b[0;31m# If we have a listlike key, _check_indexing_error will raise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'ICD10'"
     ]
    }
   ],
   "source": [
    "print(\"Not in:\")\n",
    "id_and_icd_df.loc[~id_and_icd_df[\"ICD10\"].isin(df_icd_counts_ukbb[\"ICD10\"])].drop_duplicates(subset=[\"ICD10\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "c84051da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f41202</th>\n",
       "      <th>All</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ICD10</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Q750</th>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q751</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q754</th>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q759</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q758</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       f41202  All\n",
       "ICD10             \n",
       "Q750        1    3\n",
       "Q751        2    2\n",
       "Q754        2    6\n",
       "Q759        0    2\n",
       "Q758        1    4"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df_icd_counts_ukbb.loc[df_icd_counts_ukbb[\"ICD10\"].str.startswith(\"Q0\",na=False)] # no matches for Q00\n",
    "# df_icd_counts_ukbb.loc[df_icd_counts_ukbb[\"ICD10\"].str.startswith(\"Q82\",na=False)]\n",
    "# df_icd_counts_ukbb.loc[df_icd_counts_ukbb[\"ICD10\"].str.startswith(\"N98\",na=False)]\n",
    "### N981 is probably actually N98.1 - weird coding error? ? \n",
    "\n",
    "df_icd_counts_ukbb.loc[df_icd_counts_ukbb.index.str.startswith(\"Q75\",na=False)]\n",
    "## "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a0e64ebb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ICD10</th>\n",
       "      <th>f41202</th>\n",
       "      <th>All</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1234</th>\n",
       "      <td>Q600</td>\n",
       "      <td>0</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1278</th>\n",
       "      <td>Q603</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1420</th>\n",
       "      <td>Q605</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4256</th>\n",
       "      <td>Q606</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8504</th>\n",
       "      <td>Q602</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     ICD10  f41202  All\n",
       "1234  Q600       0  109\n",
       "1278  Q603       0   10\n",
       "1420  Q605       0    1\n",
       "4256  Q606       0    2\n",
       "8504  Q602       1   31"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_icd_counts_ukbb.loc[df_icd_counts_ukbb[\"ICD10\"].str.startswith(\"Q60\",na=False)]\n",
    "## Q603 is a ICD10-CM code\n",
    "## is Q603 a code , or is it actually Q60.3 without dot?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "71b239c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "try:\n",
    "    os.mkdir(\"proc\")\n",
    "except: ()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "30158c79",
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://stackoverflow.com/questions/68814219/how-to-fix-arrowinvalid-could-not-convert-x-y-with-type-tuple"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "01782182",
   "metadata": {},
   "outputs": [],
   "source": [
    "# print(len(bad_datas))\n",
    "# print(bad_datas) # './evidence', './evidence/sourceId=europepmc', './molecule'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a2067a6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.iloc[0:2,65:67]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "35dddeaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.iloc[:,65:67].tail() # \ttargetModulation\ttextMiningSentences"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ead4efb0",
   "metadata": {},
   "source": [
    "#### munge dtypes to not use list? \n",
    "* Loses info, but easier to upload later\n",
    "* parse as str / categorical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "fdc3f76c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diseases\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>code</th>\n",
       "      <th>description</th>\n",
       "      <th>name</th>\n",
       "      <th>directLocationIds</th>\n",
       "      <th>parents</th>\n",
       "      <th>sko</th>\n",
       "      <th>synonyms</th>\n",
       "      <th>ancestors</th>\n",
       "      <th>descendants</th>\n",
       "      <th>children</th>\n",
       "      <th>therapeuticAreas</th>\n",
       "      <th>ontology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID_7551</td>\n",
       "      <td>http://purl.obolibrary.org/obo/DOID_7551</td>\n",
       "      <td>A primary bacterial infectious disease that is...</td>\n",
       "      <td>gonorrhea</td>\n",
       "      <td>None</td>\n",
       "      <td>[EFO_0003955, MONDO_0000314]</td>\n",
       "      <td>None</td>\n",
       "      <td>{'hasBroadSynonym': None, 'hasExactSynonym': [...</td>\n",
       "      <td>[EFO_0000512, EFO_0009663, MONDO_0100336, OTAR...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[MONDO_0100336, OTAR_0000017, EFO_0005741, EFO...</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': True, 'so...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HP_0000031</td>\n",
       "      <td>http://purl.obolibrary.org/obo/HP_0000031</td>\n",
       "      <td>The presence of inflammation of the epididymis.</td>\n",
       "      <td>Epididymitis</td>\n",
       "      <td>None</td>\n",
       "      <td>[HP_0012649, HP_0000022]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[HP_0000022, HP_0000078, EFO_0000651, HP_00126...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[EFO_0000651]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': True, 'so...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HP_0000857</td>\n",
       "      <td>http://purl.obolibrary.org/obo/HP_0000857</td>\n",
       "      <td>None</td>\n",
       "      <td>Neonatal insulin-dependent diabetes mellitus</td>\n",
       "      <td>None</td>\n",
       "      <td>[EFO_0000400, HP_0001952, HP_0000818]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[HP_0000818, HP_0000118, OTAR_0000020, HP_0011...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[OTAR_0000020, EFO_0000651, EFO_0001379, EFO_0...</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': True, 'so...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HP_0001693</td>\n",
       "      <td>http://purl.obolibrary.org/obo/HP_0001693</td>\n",
       "      <td>Pattern of blood flow in the heart that deviat...</td>\n",
       "      <td>Cardiac shunt</td>\n",
       "      <td>None</td>\n",
       "      <td>[HP_0011028]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[EFO_0000651, HP_0011028, HP_0000118, HP_00110...</td>\n",
       "      <td>[HP_0012382]</td>\n",
       "      <td>[HP_0012382]</td>\n",
       "      <td>[EFO_0000651]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': False, 's...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HP_0001730</td>\n",
       "      <td>http://purl.obolibrary.org/obo/HP_0001730</td>\n",
       "      <td>A progressive form of hearing impairment.</td>\n",
       "      <td>Progressive hearing impairment</td>\n",
       "      <td>None</td>\n",
       "      <td>[HP_0000364]</td>\n",
       "      <td>None</td>\n",
       "      <td>{'hasBroadSynonym': None, 'hasExactSynonym': [...</td>\n",
       "      <td>[HP_0000364, EFO_0000651, HP_0000598, HP_0000118]</td>\n",
       "      <td>[HP_0000408]</td>\n",
       "      <td>[HP_0000408]</td>\n",
       "      <td>[EFO_0000651]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': False, 's...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23069</th>\n",
       "      <td>Orphanet_721</td>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_721</td>\n",
       "      <td>Gray platelet syndrome (GPS) is a rare inherit...</td>\n",
       "      <td>Gray platelet syndrome</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_98455]</td>\n",
       "      <td>None</td>\n",
       "      <td>{'hasBroadSynonym': None, 'hasExactSynonym': [...</td>\n",
       "      <td>[Orphanet_183654, OTAR_0000018, Orphanet_98454...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[OTAR_0000018, EFO_0005803]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': True, 'so...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23070</th>\n",
       "      <td>Orphanet_93322</td>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_93322</td>\n",
       "      <td>Tibial hemimelia is a rare congenital anomaly ...</td>\n",
       "      <td>Tibial hemimelia</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_2130]</td>\n",
       "      <td>None</td>\n",
       "      <td>{'hasBroadSynonym': None, 'hasExactSynonym': [...</td>\n",
       "      <td>[EFO_0004260, Orphanet_404571, OTAR_0000006, O...</td>\n",
       "      <td>[Orphanet_295079, Orphanet_295077]</td>\n",
       "      <td>[Orphanet_295077, Orphanet_295079]</td>\n",
       "      <td>[OTAR_0000006, OTAR_0000018]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': False, 's...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23071</th>\n",
       "      <td>Orphanet_93925</td>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_93925</td>\n",
       "      <td>Alobar holoprosencephaly is the most severe cl...</td>\n",
       "      <td>Alobar holoprosencephaly</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_2162]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_183763, Orphanet_183533, OTAR_000001...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[OTAR_0000018, EFO_0001379]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': True, 'so...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23072</th>\n",
       "      <td>Orphanet_98539</td>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_98539</td>\n",
       "      <td>None</td>\n",
       "      <td>Early-onset ataxia with dementia</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_98538]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_71859, Orphanet_183500, Orphanet_276...</td>\n",
       "      <td>[Orphanet_247815, EFO_0009016, EFO_0010249, EF...</td>\n",
       "      <td>[Orphanet_1172]</td>\n",
       "      <td>[OTAR_0000018, EFO_0000618]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': False, 's...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23073</th>\n",
       "      <td>Orphanet_98636</td>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_98636</td>\n",
       "      <td>None</td>\n",
       "      <td>Corneoiridogoniodysgenesis</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_98632]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[Orphanet_359, Orphanet_101435, OTAR_0000018, ...</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[OTAR_0000018, MONDO_0024458]</td>\n",
       "      <td>{'isTherapeuticArea': False, 'leaf': True, 'so...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23074 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   id                                       code  \\\n",
       "0           DOID_7551   http://purl.obolibrary.org/obo/DOID_7551   \n",
       "1          HP_0000031  http://purl.obolibrary.org/obo/HP_0000031   \n",
       "2          HP_0000857  http://purl.obolibrary.org/obo/HP_0000857   \n",
       "3          HP_0001693  http://purl.obolibrary.org/obo/HP_0001693   \n",
       "4          HP_0001730  http://purl.obolibrary.org/obo/HP_0001730   \n",
       "...               ...                                        ...   \n",
       "23069    Orphanet_721     http://www.orpha.net/ORDO/Orphanet_721   \n",
       "23070  Orphanet_93322   http://www.orpha.net/ORDO/Orphanet_93322   \n",
       "23071  Orphanet_93925   http://www.orpha.net/ORDO/Orphanet_93925   \n",
       "23072  Orphanet_98539   http://www.orpha.net/ORDO/Orphanet_98539   \n",
       "23073  Orphanet_98636   http://www.orpha.net/ORDO/Orphanet_98636   \n",
       "\n",
       "                                             description  \\\n",
       "0      A primary bacterial infectious disease that is...   \n",
       "1        The presence of inflammation of the epididymis.   \n",
       "2                                                   None   \n",
       "3      Pattern of blood flow in the heart that deviat...   \n",
       "4              A progressive form of hearing impairment.   \n",
       "...                                                  ...   \n",
       "23069  Gray platelet syndrome (GPS) is a rare inherit...   \n",
       "23070  Tibial hemimelia is a rare congenital anomaly ...   \n",
       "23071  Alobar holoprosencephaly is the most severe cl...   \n",
       "23072                                               None   \n",
       "23073                                               None   \n",
       "\n",
       "                                               name directLocationIds  \\\n",
       "0                                         gonorrhea              None   \n",
       "1                                      Epididymitis              None   \n",
       "2      Neonatal insulin-dependent diabetes mellitus              None   \n",
       "3                                     Cardiac shunt              None   \n",
       "4                    Progressive hearing impairment              None   \n",
       "...                                             ...               ...   \n",
       "23069                        Gray platelet syndrome              None   \n",
       "23070                              Tibial hemimelia              None   \n",
       "23071                      Alobar holoprosencephaly              None   \n",
       "23072              Early-onset ataxia with dementia              None   \n",
       "23073                    Corneoiridogoniodysgenesis              None   \n",
       "\n",
       "                                     parents   sko  \\\n",
       "0               [EFO_0003955, MONDO_0000314]  None   \n",
       "1                   [HP_0012649, HP_0000022]  None   \n",
       "2      [EFO_0000400, HP_0001952, HP_0000818]  None   \n",
       "3                               [HP_0011028]  None   \n",
       "4                               [HP_0000364]  None   \n",
       "...                                      ...   ...   \n",
       "23069                       [Orphanet_98455]  None   \n",
       "23070                        [Orphanet_2130]  None   \n",
       "23071                        [Orphanet_2162]  None   \n",
       "23072                       [Orphanet_98538]  None   \n",
       "23073                       [Orphanet_98632]  None   \n",
       "\n",
       "                                                synonyms  \\\n",
       "0      {'hasBroadSynonym': None, 'hasExactSynonym': [...   \n",
       "1                                                   None   \n",
       "2                                                   None   \n",
       "3                                                   None   \n",
       "4      {'hasBroadSynonym': None, 'hasExactSynonym': [...   \n",
       "...                                                  ...   \n",
       "23069  {'hasBroadSynonym': None, 'hasExactSynonym': [...   \n",
       "23070  {'hasBroadSynonym': None, 'hasExactSynonym': [...   \n",
       "23071                                               None   \n",
       "23072                                               None   \n",
       "23073                                               None   \n",
       "\n",
       "                                               ancestors  \\\n",
       "0      [EFO_0000512, EFO_0009663, MONDO_0100336, OTAR...   \n",
       "1      [HP_0000022, HP_0000078, EFO_0000651, HP_00126...   \n",
       "2      [HP_0000818, HP_0000118, OTAR_0000020, HP_0011...   \n",
       "3      [EFO_0000651, HP_0011028, HP_0000118, HP_00110...   \n",
       "4      [HP_0000364, EFO_0000651, HP_0000598, HP_0000118]   \n",
       "...                                                  ...   \n",
       "23069  [Orphanet_183654, OTAR_0000018, Orphanet_98454...   \n",
       "23070  [EFO_0004260, Orphanet_404571, OTAR_0000006, O...   \n",
       "23071  [Orphanet_183763, Orphanet_183533, OTAR_000001...   \n",
       "23072  [Orphanet_71859, Orphanet_183500, Orphanet_276...   \n",
       "23073  [Orphanet_359, Orphanet_101435, OTAR_0000018, ...   \n",
       "\n",
       "                                             descendants  \\\n",
       "0                                                     []   \n",
       "1                                                     []   \n",
       "2                                                     []   \n",
       "3                                           [HP_0012382]   \n",
       "4                                           [HP_0000408]   \n",
       "...                                                  ...   \n",
       "23069                                                 []   \n",
       "23070                 [Orphanet_295079, Orphanet_295077]   \n",
       "23071                                                 []   \n",
       "23072  [Orphanet_247815, EFO_0009016, EFO_0010249, EF...   \n",
       "23073                                                 []   \n",
       "\n",
       "                                 children  \\\n",
       "0                                      []   \n",
       "1                                      []   \n",
       "2                                      []   \n",
       "3                            [HP_0012382]   \n",
       "4                            [HP_0000408]   \n",
       "...                                   ...   \n",
       "23069                                  []   \n",
       "23070  [Orphanet_295077, Orphanet_295079]   \n",
       "23071                                  []   \n",
       "23072                     [Orphanet_1172]   \n",
       "23073                                  []   \n",
       "\n",
       "                                        therapeuticAreas  \\\n",
       "0      [MONDO_0100336, OTAR_0000017, EFO_0005741, EFO...   \n",
       "1                                          [EFO_0000651]   \n",
       "2      [OTAR_0000020, EFO_0000651, EFO_0001379, EFO_0...   \n",
       "3                                          [EFO_0000651]   \n",
       "4                                          [EFO_0000651]   \n",
       "...                                                  ...   \n",
       "23069                        [OTAR_0000018, EFO_0005803]   \n",
       "23070                       [OTAR_0000006, OTAR_0000018]   \n",
       "23071                        [OTAR_0000018, EFO_0001379]   \n",
       "23072                        [OTAR_0000018, EFO_0000618]   \n",
       "23073                      [OTAR_0000018, MONDO_0024458]   \n",
       "\n",
       "                                                ontology  \n",
       "0      {'isTherapeuticArea': False, 'leaf': True, 'so...  \n",
       "1      {'isTherapeuticArea': False, 'leaf': True, 'so...  \n",
       "2      {'isTherapeuticArea': False, 'leaf': True, 'so...  \n",
       "3      {'isTherapeuticArea': False, 'leaf': False, 's...  \n",
       "4      {'isTherapeuticArea': False, 'leaf': False, 's...  \n",
       "...                                                  ...  \n",
       "23069  {'isTherapeuticArea': False, 'leaf': True, 'so...  \n",
       "23070  {'isTherapeuticArea': False, 'leaf': False, 's...  \n",
       "23071  {'isTherapeuticArea': False, 'leaf': True, 'so...  \n",
       "23072  {'isTherapeuticArea': False, 'leaf': False, 's...  \n",
       "23073  {'isTherapeuticArea': False, 'leaf': True, 'so...  \n",
       "\n",
       "[23074 rows x 13 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23074\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 23074 entries, 0 to 23073\n",
      "Data columns (total 13 columns):\n",
      " #   Column             Non-Null Count  Dtype   \n",
      "---  ------             --------------  -----   \n",
      " 0   id                 23074 non-null  object  \n",
      " 1   code               23074 non-null  object  \n",
      " 2   description        16949 non-null  object  \n",
      " 3   name               23074 non-null  object  \n",
      " 4   directLocationIds  12 non-null     category\n",
      " 5   parents            23074 non-null  object  \n",
      " 6   sko                1 non-null      category\n",
      " 7   synonyms           16339 non-null  object  \n",
      " 8   ancestors          23074 non-null  object  \n",
      " 9   descendants        23074 non-null  object  \n",
      " 10  children           23074 non-null  object  \n",
      " 11  therapeuticAreas   23074 non-null  category\n",
      " 12  ontology           23074 non-null  object  \n",
      "dtypes: category(3), object(10)\n",
      "memory usage: 2.1+ MB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diseaseToPhenotype\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>evidence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EFO_0000182</td>\n",
       "      <td>HP_0001413</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'HPO:probinson...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EFO_0000222</td>\n",
       "      <td>HP_0000006</td>\n",
       "      <td>[{'aspect': 'I', 'bioCuration': 'HPO:skoehler[...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EFO_0000239</td>\n",
       "      <td>HP_0003574</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'HPO:iea[2009-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EFO_0000272</td>\n",
       "      <td>HP_0009592</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'HPO:skoehler[...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EFO_0000274</td>\n",
       "      <td>HP_0001041</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'HPO:iea[2009-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206499</th>\n",
       "      <td>Orphanet_99939</td>\n",
       "      <td>HP_0003431</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'HPO:probinson...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206500</th>\n",
       "      <td>Orphanet_99943</td>\n",
       "      <td>HP_0000408</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'HPO:iea[2009-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206501</th>\n",
       "      <td>Orphanet_99944</td>\n",
       "      <td>HP_0001288</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'ORPHA:orphada...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206502</th>\n",
       "      <td>Orphanet_99947</td>\n",
       "      <td>HP_0001609</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'ORPHA:orphada...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206503</th>\n",
       "      <td>Orphanet_99947</td>\n",
       "      <td>HP_0002936</td>\n",
       "      <td>[{'aspect': 'P', 'bioCuration': 'HPO:skoehler[...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>206504 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               disease   phenotype  \\\n",
       "0          EFO_0000182  HP_0001413   \n",
       "1          EFO_0000222  HP_0000006   \n",
       "2          EFO_0000239  HP_0003574   \n",
       "3          EFO_0000272  HP_0009592   \n",
       "4          EFO_0000274  HP_0001041   \n",
       "...                ...         ...   \n",
       "206499  Orphanet_99939  HP_0003431   \n",
       "206500  Orphanet_99943  HP_0000408   \n",
       "206501  Orphanet_99944  HP_0001288   \n",
       "206502  Orphanet_99947  HP_0001609   \n",
       "206503  Orphanet_99947  HP_0002936   \n",
       "\n",
       "                                                 evidence  \n",
       "0       [{'aspect': 'P', 'bioCuration': 'HPO:probinson...  \n",
       "1       [{'aspect': 'I', 'bioCuration': 'HPO:skoehler[...  \n",
       "2       [{'aspect': 'P', 'bioCuration': 'HPO:iea[2009-...  \n",
       "3       [{'aspect': 'P', 'bioCuration': 'HPO:skoehler[...  \n",
       "4       [{'aspect': 'P', 'bioCuration': 'HPO:iea[2009-...  \n",
       "...                                                   ...  \n",
       "206499  [{'aspect': 'P', 'bioCuration': 'HPO:probinson...  \n",
       "206500  [{'aspect': 'P', 'bioCuration': 'HPO:iea[2009-...  \n",
       "206501  [{'aspect': 'P', 'bioCuration': 'ORPHA:orphada...  \n",
       "206502  [{'aspect': 'P', 'bioCuration': 'ORPHA:orphada...  \n",
       "206503  [{'aspect': 'P', 'bioCuration': 'HPO:skoehler[...  \n",
       "\n",
       "[206504 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "206504\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 206504 entries, 0 to 206503\n",
      "Data columns (total 3 columns):\n",
      " #   Column     Non-Null Count   Dtype   \n",
      "---  ------     --------------   -----   \n",
      " 0   disease    206504 non-null  category\n",
      " 1   phenotype  206504 non-null  category\n",
      " 2   evidence   206504 non-null  object  \n",
      "dtypes: category(2), object(1)\n",
      "memory usage: 4.6+ MB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "go\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:0000001</td>\n",
       "      <td>mitochondrion inheritance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:0000002</td>\n",
       "      <td>mitochondrial genome maintenance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:0000003</td>\n",
       "      <td>reproduction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:0000005</td>\n",
       "      <td>obsolete ribosomal chaperone activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GO:0000006</td>\n",
       "      <td>high-affinity zinc transmembrane transporter a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47300</th>\n",
       "      <td>GO:2001313</td>\n",
       "      <td>UDP-4-deoxy-4-formamido-beta-L-arabinopyranose...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47301</th>\n",
       "      <td>GO:2001314</td>\n",
       "      <td>UDP-4-deoxy-4-formamido-beta-L-arabinopyranose...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47302</th>\n",
       "      <td>GO:2001315</td>\n",
       "      <td>UDP-4-deoxy-4-formamido-beta-L-arabinopyranose...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47303</th>\n",
       "      <td>GO:2001316</td>\n",
       "      <td>kojic acid metabolic process</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47304</th>\n",
       "      <td>GO:2001317</td>\n",
       "      <td>kojic acid biosynthetic process</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47305 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               id                                               name\n",
       "0      GO:0000001                          mitochondrion inheritance\n",
       "1      GO:0000002                   mitochondrial genome maintenance\n",
       "2      GO:0000003                                       reproduction\n",
       "3      GO:0000005              obsolete ribosomal chaperone activity\n",
       "4      GO:0000006  high-affinity zinc transmembrane transporter a...\n",
       "...           ...                                                ...\n",
       "47300  GO:2001313  UDP-4-deoxy-4-formamido-beta-L-arabinopyranose...\n",
       "47301  GO:2001314  UDP-4-deoxy-4-formamido-beta-L-arabinopyranose...\n",
       "47302  GO:2001315  UDP-4-deoxy-4-formamido-beta-L-arabinopyranose...\n",
       "47303  GO:2001316                       kojic acid metabolic process\n",
       "47304  GO:2001317                    kojic acid biosynthetic process\n",
       "\n",
       "[47305 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "47305\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 47305 entries, 0 to 47304\n",
      "Data columns (total 2 columns):\n",
      " #   Column  Non-Null Count  Dtype \n",
      "---  ------  --------------  ----- \n",
      " 0   id      47305 non-null  object\n",
      " 1   name    47305 non-null  object\n",
      "dtypes: object(2)\n",
      "memory usage: 1.1+ MB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "targets\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>approvedSymbol</th>\n",
       "      <th>biotype</th>\n",
       "      <th>transcriptIds</th>\n",
       "      <th>genomicLocation</th>\n",
       "      <th>alternativeGenes</th>\n",
       "      <th>approvedName</th>\n",
       "      <th>go</th>\n",
       "      <th>hallmarks</th>\n",
       "      <th>synonyms</th>\n",
       "      <th>...</th>\n",
       "      <th>targetClass</th>\n",
       "      <th>constraint</th>\n",
       "      <th>tep</th>\n",
       "      <th>proteinIds</th>\n",
       "      <th>dbXrefs</th>\n",
       "      <th>chemicalProbes</th>\n",
       "      <th>homologues</th>\n",
       "      <th>tractability</th>\n",
       "      <th>safetyLiabilities</th>\n",
       "      <th>pathways</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000020219</td>\n",
       "      <td>CCT8L1P</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>[ENST00000465400]</td>\n",
       "      <td>{'chromosome': '7', 'start': 152445477, 'end':...</td>\n",
       "      <td>None</td>\n",
       "      <td>chaperonin containing TCP1 subunit 8 like 1, p...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'label': 'Putative T-complex protein 1 subun...</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'id': 'A6NM43', 'source': 'uniprot_obsolete'}]</td>\n",
       "      <td>[{'id': '32153', 'source': 'HGNC'}, {'id': 'IP...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000059588</td>\n",
       "      <td>TARBP1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>[ENST00000496673, ENST00000483404, ENST0000048...</td>\n",
       "      <td>{'chromosome': '1', 'start': 234391313, 'end':...</td>\n",
       "      <td>None</td>\n",
       "      <td>TAR (HIV-1) RNA binding protein 1</td>\n",
       "      <td>[{'id': 'GO:0016423', 'source': 'PMID:21873635...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'label': 'Probable methyltransferase TARBP1'...</td>\n",
       "      <td>...</td>\n",
       "      <td>[{'id': 5, 'label': 'Other nuclear protein', '...</td>\n",
       "      <td>[{'constraintType': 'syn', 'score': 0.88121002...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'id': 'Q13395', 'source': 'uniprot_swissprot...</td>\n",
       "      <td>[{'id': '11568', 'source': 'HGNC'}, {'id': '2H...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'speciesId': '9606', 'speciesName': 'Human',...</td>\n",
       "      <td>[{'modality': 'SM', 'id': 'Approved Drug', 'va...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000070182</td>\n",
       "      <td>SPTB</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>[ENST00000553938, ENST00000389720, ENST0000055...</td>\n",
       "      <td>{'chromosome': '14', 'start': 64746283, 'end':...</td>\n",
       "      <td>None</td>\n",
       "      <td>spectrin beta, erythrocytic</td>\n",
       "      <td>[{'id': 'GO:0030506', 'source': 'PMID:8159688'...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'label': 'Spectrin beta chain, erythrocytic'...</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'constraintType': 'syn', 'score': -2.2815999...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'id': 'P11277', 'source': 'uniprot_swissprot...</td>\n",
       "      <td>[{'id': '11274', 'source': 'HGNC'}, {'id': '1S...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'speciesId': '9606', 'speciesName': 'Human',...</td>\n",
       "      <td>[{'modality': 'SM', 'id': 'Approved Drug', 'va...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'pathwayId': 'R-HSA-6807878', 'pathway': 'CO...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000070366</td>\n",
       "      <td>SMG6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>[ENST00000354901, ENST00000570756, ENST0000026...</td>\n",
       "      <td>{'chromosome': '17', 'start': 2059839, 'end': ...</td>\n",
       "      <td>None</td>\n",
       "      <td>SMG6 nonsense mediated mRNA decay factor</td>\n",
       "      <td>[{'id': 'GO:1904354', 'source': 'PMID:12676087...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'label': 'Telomerase-binding protein EST1A',...</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'constraintType': 'syn', 'score': -2.2750000...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'id': 'Q86US8', 'source': 'uniprot_swissprot...</td>\n",
       "      <td>[{'id': '17809', 'source': 'HGNC'}, {'id': '2D...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'speciesId': '9598', 'speciesName': 'Chimpan...</td>\n",
       "      <td>[{'modality': 'SM', 'id': 'Approved Drug', 'va...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'pathwayId': 'R-HSA-975957', 'pathway': 'Non...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000072071</td>\n",
       "      <td>ADGRL1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>[ENST00000361434, ENST00000589616, ENST0000059...</td>\n",
       "      <td>{'chromosome': '19', 'start': 14147743, 'end':...</td>\n",
       "      <td>[ENSG00000288324]</td>\n",
       "      <td>adhesion G protein-coupled receptor L1</td>\n",
       "      <td>[{'id': 'GO:0016524', 'source': 'GO_REF:000002...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'label': 'Adhesion G protein-coupled recepto...</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'constraintType': 'syn', 'score': -1.7188999...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'id': 'O94910', 'source': 'uniprot_swissprot...</td>\n",
       "      <td>[{'id': '20973', 'source': 'HGNC'}, {'id': 'IP...</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'speciesId': '9606', 'speciesName': 'Human',...</td>\n",
       "      <td>[{'modality': 'SM', 'id': 'Approved Drug', 'va...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61519</th>\n",
       "      <td>ENSG00000289308</td>\n",
       "      <td>ENSG00000289308</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>[ENST00000686538, ENST00000691227, ENST0000068...</td>\n",
       "      <td>{'chromosome': '4', 'start': 61749388, 'end': ...</td>\n",
       "      <td>None</td>\n",
       "      <td>novel transcript, antisense to ADGRL3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61520</th>\n",
       "      <td>ENSG00000289437</td>\n",
       "      <td>ENSG00000289437</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>[ENST00000692710]</td>\n",
       "      <td>{'chromosome': '12', 'start': 95260099, 'end':...</td>\n",
       "      <td>None</td>\n",
       "      <td>novel transcript, sense intronic to VEZT</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61521</th>\n",
       "      <td>ENSG00000289492</td>\n",
       "      <td>ENSG00000289492</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>[ENST00000692062]</td>\n",
       "      <td>{'chromosome': '20', 'start': 31738617, 'end':...</td>\n",
       "      <td>None</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61522</th>\n",
       "      <td>ENSG00000289511</td>\n",
       "      <td>ENSG00000289511</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>[ENST00000690385]</td>\n",
       "      <td>{'chromosome': '21', 'start': 46667379, 'end':...</td>\n",
       "      <td>None</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61523</th>\n",
       "      <td>ENSG00000289522</td>\n",
       "      <td>ENSG00000289522</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>[ENST00000685955]</td>\n",
       "      <td>{'chromosome': '15', 'start': 25636451, 'end':...</td>\n",
       "      <td>None</td>\n",
       "      <td>novel transcript</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'label': 'LOC124903448', 'source': 'NCBI_ent...</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61524 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id   approvedSymbol               biotype  \\\n",
       "0      ENSG00000020219          CCT8L1P  processed_pseudogene   \n",
       "1      ENSG00000059588           TARBP1        protein_coding   \n",
       "2      ENSG00000070182             SPTB        protein_coding   \n",
       "3      ENSG00000070366             SMG6        protein_coding   \n",
       "4      ENSG00000072071           ADGRL1        protein_coding   \n",
       "...                ...              ...                   ...   \n",
       "61519  ENSG00000289308  ENSG00000289308                lncRNA   \n",
       "61520  ENSG00000289437  ENSG00000289437                lncRNA   \n",
       "61521  ENSG00000289492  ENSG00000289492                lncRNA   \n",
       "61522  ENSG00000289511  ENSG00000289511                lncRNA   \n",
       "61523  ENSG00000289522  ENSG00000289522                lncRNA   \n",
       "\n",
       "                                           transcriptIds  \\\n",
       "0                                      [ENST00000465400]   \n",
       "1      [ENST00000496673, ENST00000483404, ENST0000048...   \n",
       "2      [ENST00000553938, ENST00000389720, ENST0000055...   \n",
       "3      [ENST00000354901, ENST00000570756, ENST0000026...   \n",
       "4      [ENST00000361434, ENST00000589616, ENST0000059...   \n",
       "...                                                  ...   \n",
       "61519  [ENST00000686538, ENST00000691227, ENST0000068...   \n",
       "61520                                  [ENST00000692710]   \n",
       "61521                                  [ENST00000692062]   \n",
       "61522                                  [ENST00000690385]   \n",
       "61523                                  [ENST00000685955]   \n",
       "\n",
       "                                         genomicLocation   alternativeGenes  \\\n",
       "0      {'chromosome': '7', 'start': 152445477, 'end':...               None   \n",
       "1      {'chromosome': '1', 'start': 234391313, 'end':...               None   \n",
       "2      {'chromosome': '14', 'start': 64746283, 'end':...               None   \n",
       "3      {'chromosome': '17', 'start': 2059839, 'end': ...               None   \n",
       "4      {'chromosome': '19', 'start': 14147743, 'end':...  [ENSG00000288324]   \n",
       "...                                                  ...                ...   \n",
       "61519  {'chromosome': '4', 'start': 61749388, 'end': ...               None   \n",
       "61520  {'chromosome': '12', 'start': 95260099, 'end':...               None   \n",
       "61521  {'chromosome': '20', 'start': 31738617, 'end':...               None   \n",
       "61522  {'chromosome': '21', 'start': 46667379, 'end':...               None   \n",
       "61523  {'chromosome': '15', 'start': 25636451, 'end':...               None   \n",
       "\n",
       "                                            approvedName  \\\n",
       "0      chaperonin containing TCP1 subunit 8 like 1, p...   \n",
       "1                      TAR (HIV-1) RNA binding protein 1   \n",
       "2                            spectrin beta, erythrocytic   \n",
       "3               SMG6 nonsense mediated mRNA decay factor   \n",
       "4                 adhesion G protein-coupled receptor L1   \n",
       "...                                                  ...   \n",
       "61519              novel transcript, antisense to ADGRL3   \n",
       "61520           novel transcript, sense intronic to VEZT   \n",
       "61521                                   novel transcript   \n",
       "61522                                   novel transcript   \n",
       "61523                                   novel transcript   \n",
       "\n",
       "                                                      go hallmarks  \\\n",
       "0                                                   None      None   \n",
       "1      [{'id': 'GO:0016423', 'source': 'PMID:21873635...      None   \n",
       "2      [{'id': 'GO:0030506', 'source': 'PMID:8159688'...      None   \n",
       "3      [{'id': 'GO:1904354', 'source': 'PMID:12676087...      None   \n",
       "4      [{'id': 'GO:0016524', 'source': 'GO_REF:000002...      None   \n",
       "...                                                  ...       ...   \n",
       "61519                                               None      None   \n",
       "61520                                               None      None   \n",
       "61521                                               None      None   \n",
       "61522                                               None      None   \n",
       "61523                                               None      None   \n",
       "\n",
       "                                                synonyms  ...  \\\n",
       "0      [{'label': 'Putative T-complex protein 1 subun...  ...   \n",
       "1      [{'label': 'Probable methyltransferase TARBP1'...  ...   \n",
       "2      [{'label': 'Spectrin beta chain, erythrocytic'...  ...   \n",
       "3      [{'label': 'Telomerase-binding protein EST1A',...  ...   \n",
       "4      [{'label': 'Adhesion G protein-coupled recepto...  ...   \n",
       "...                                                  ...  ...   \n",
       "61519                                                 []  ...   \n",
       "61520                                                 []  ...   \n",
       "61521                                                 []  ...   \n",
       "61522                                                 []  ...   \n",
       "61523  [{'label': 'LOC124903448', 'source': 'NCBI_ent...  ...   \n",
       "\n",
       "                                             targetClass  \\\n",
       "0                                                   None   \n",
       "1      [{'id': 5, 'label': 'Other nuclear protein', '...   \n",
       "2                                                   None   \n",
       "3                                                   None   \n",
       "4                                                   None   \n",
       "...                                                  ...   \n",
       "61519                                               None   \n",
       "61520                                               None   \n",
       "61521                                               None   \n",
       "61522                                               None   \n",
       "61523                                               None   \n",
       "\n",
       "                                              constraint   tep  \\\n",
       "0                                                   None  None   \n",
       "1      [{'constraintType': 'syn', 'score': 0.88121002...  None   \n",
       "2      [{'constraintType': 'syn', 'score': -2.2815999...  None   \n",
       "3      [{'constraintType': 'syn', 'score': -2.2750000...  None   \n",
       "4      [{'constraintType': 'syn', 'score': -1.7188999...  None   \n",
       "...                                                  ...   ...   \n",
       "61519                                               None  None   \n",
       "61520                                               None  None   \n",
       "61521                                               None  None   \n",
       "61522                                               None  None   \n",
       "61523                                               None  None   \n",
       "\n",
       "                                              proteinIds  \\\n",
       "0       [{'id': 'A6NM43', 'source': 'uniprot_obsolete'}]   \n",
       "1      [{'id': 'Q13395', 'source': 'uniprot_swissprot...   \n",
       "2      [{'id': 'P11277', 'source': 'uniprot_swissprot...   \n",
       "3      [{'id': 'Q86US8', 'source': 'uniprot_swissprot...   \n",
       "4      [{'id': 'O94910', 'source': 'uniprot_swissprot...   \n",
       "...                                                  ...   \n",
       "61519                                               None   \n",
       "61520                                               None   \n",
       "61521                                               None   \n",
       "61522                                               None   \n",
       "61523                                               None   \n",
       "\n",
       "                                                 dbXrefs chemicalProbes  \\\n",
       "0      [{'id': '32153', 'source': 'HGNC'}, {'id': 'IP...           None   \n",
       "1      [{'id': '11568', 'source': 'HGNC'}, {'id': '2H...           None   \n",
       "2      [{'id': '11274', 'source': 'HGNC'}, {'id': '1S...           None   \n",
       "3      [{'id': '17809', 'source': 'HGNC'}, {'id': '2D...           None   \n",
       "4      [{'id': '20973', 'source': 'HGNC'}, {'id': 'IP...           None   \n",
       "...                                                  ...            ...   \n",
       "61519                                                 []           None   \n",
       "61520                                                 []           None   \n",
       "61521                                                 []           None   \n",
       "61522                                                 []           None   \n",
       "61523                                                 []           None   \n",
       "\n",
       "                                              homologues  \\\n",
       "0                                                   None   \n",
       "1      [{'speciesId': '9606', 'speciesName': 'Human',...   \n",
       "2      [{'speciesId': '9606', 'speciesName': 'Human',...   \n",
       "3      [{'speciesId': '9598', 'speciesName': 'Chimpan...   \n",
       "4      [{'speciesId': '9606', 'speciesName': 'Human',...   \n",
       "...                                                  ...   \n",
       "61519                                               None   \n",
       "61520                                               None   \n",
       "61521                                               None   \n",
       "61522                                               None   \n",
       "61523                                               None   \n",
       "\n",
       "                                            tractability safetyLiabilities  \\\n",
       "0                                                   None              None   \n",
       "1      [{'modality': 'SM', 'id': 'Approved Drug', 'va...              None   \n",
       "2      [{'modality': 'SM', 'id': 'Approved Drug', 'va...              None   \n",
       "3      [{'modality': 'SM', 'id': 'Approved Drug', 'va...              None   \n",
       "4      [{'modality': 'SM', 'id': 'Approved Drug', 'va...              None   \n",
       "...                                                  ...               ...   \n",
       "61519                                               None              None   \n",
       "61520                                               None              None   \n",
       "61521                                               None              None   \n",
       "61522                                               None              None   \n",
       "61523                                               None              None   \n",
       "\n",
       "                                                pathways  \n",
       "0                                                   None  \n",
       "1                                                   None  \n",
       "2      [{'pathwayId': 'R-HSA-6807878', 'pathway': 'CO...  \n",
       "3      [{'pathwayId': 'R-HSA-975957', 'pathway': 'Non...  \n",
       "4                                                   None  \n",
       "...                                                  ...  \n",
       "61519                                               None  \n",
       "61520                                               None  \n",
       "61521                                               None  \n",
       "61522                                               None  \n",
       "61523                                               None  \n",
       "\n",
       "[61524 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "61524\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 61524 entries, 0 to 61523\n",
      "Data columns (total 24 columns):\n",
      " #   Column                Non-Null Count  Dtype   \n",
      "---  ------                --------------  -----   \n",
      " 0   id                    61524 non-null  object  \n",
      " 1   approvedSymbol        61524 non-null  object  \n",
      " 2   biotype               61524 non-null  category\n",
      " 3   transcriptIds         61524 non-null  object  \n",
      " 4   genomicLocation       61524 non-null  object  \n",
      " 5   alternativeGenes      1081 non-null   category\n",
      " 6   approvedName          61524 non-null  object  \n",
      " 7   go                    20947 non-null  object  \n",
      " 8   hallmarks             351 non-null    category\n",
      " 9   synonyms              61524 non-null  object  \n",
      " 10  symbolSynonyms        61524 non-null  object  \n",
      " 11  nameSynonyms          61524 non-null  object  \n",
      " 12  functionDescriptions  20139 non-null  object  \n",
      " 13  subcellularLocations  20139 non-null  object  \n",
      " 14  targetClass           4607 non-null   category\n",
      " 15  constraint            19116 non-null  object  \n",
      " 16  tep                   41 non-null     category\n",
      " 17  proteinIds            21071 non-null  object  \n",
      " 18  dbXrefs               61524 non-null  object  \n",
      " 19  chemicalProbes        634 non-null    category\n",
      " 20  homologues            25964 non-null  object  \n",
      " 21  tractability          19480 non-null  category\n",
      " 22  safetyLiabilities     456 non-null    category\n",
      " 23  pathways              11003 non-null  category\n",
      "dtypes: category(9), object(15)\n",
      "memory usage: 8.8+ MB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hpo\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>namespace</th>\n",
       "      <th>parents</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>BFO_0000019</td>\n",
       "      <td>quality</td>\n",
       "      <td>[bfo]</td>\n",
       "      <td>[b91618, BFO_0000020]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>BFO_0000030</td>\n",
       "      <td>object</td>\n",
       "      <td>[bfo]</td>\n",
       "      <td>[BFO_0000040]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>An independent continuant that is spatially ex...</td>\n",
       "      <td>BFO_0000040</td>\n",
       "      <td>material entity</td>\n",
       "      <td>[bfo]</td>\n",
       "      <td>[BFO_0000004]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Elementary particle not affected by the strong...</td>\n",
       "      <td>CHEBI_10545</td>\n",
       "      <td>electron</td>\n",
       "      <td>[chebi_ontology]</td>\n",
       "      <td>[CHEBI_36338]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A dicarboxylic acid dianion resulting from the...</td>\n",
       "      <td>CHEBI_11596</td>\n",
       "      <td>2-hydroxyglutarate(2-)</td>\n",
       "      <td>[chebi_ontology]</td>\n",
       "      <td>[b67169, CHEBI_28965, CHEBI_18059, b67170, CHE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26365</th>\n",
       "      <td>A protein coding gene OPLAH in human.</td>\n",
       "      <td>HGNC_8149</td>\n",
       "      <td>OPLAH (human)</td>\n",
       "      <td>[gene]</td>\n",
       "      <td>[b64807, SO_0001217]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26366</th>\n",
       "      <td>A protein coding gene PAH in human.</td>\n",
       "      <td>HGNC_8582</td>\n",
       "      <td>PAH (human)</td>\n",
       "      <td>[gene]</td>\n",
       "      <td>[b67084, SO_0001217]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26367</th>\n",
       "      <td>A protein coding gene PRL in human.</td>\n",
       "      <td>HGNC_9445</td>\n",
       "      <td>PRL (human)</td>\n",
       "      <td>[gene]</td>\n",
       "      <td>[b133804, SO_0001217]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26368</th>\n",
       "      <td>A protein coding gene PTH in human.</td>\n",
       "      <td>HGNC_9606</td>\n",
       "      <td>PTH (human)</td>\n",
       "      <td>[gene]</td>\n",
       "      <td>[b141039, SO_0001217]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26369</th>\n",
       "      <td>A protein coding gene REN in human.</td>\n",
       "      <td>HGNC_9958</td>\n",
       "      <td>REN (human)</td>\n",
       "      <td>[gene]</td>\n",
       "      <td>[b3505, SO_0001217]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26370 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             description           id  \\\n",
       "0                                                   None  BFO_0000019   \n",
       "1                                                   None  BFO_0000030   \n",
       "2      An independent continuant that is spatially ex...  BFO_0000040   \n",
       "3      Elementary particle not affected by the strong...  CHEBI_10545   \n",
       "4      A dicarboxylic acid dianion resulting from the...  CHEBI_11596   \n",
       "...                                                  ...          ...   \n",
       "26365              A protein coding gene OPLAH in human.    HGNC_8149   \n",
       "26366                A protein coding gene PAH in human.    HGNC_8582   \n",
       "26367                A protein coding gene PRL in human.    HGNC_9445   \n",
       "26368                A protein coding gene PTH in human.    HGNC_9606   \n",
       "26369                A protein coding gene REN in human.    HGNC_9958   \n",
       "\n",
       "                         name         namespace  \\\n",
       "0                     quality             [bfo]   \n",
       "1                      object             [bfo]   \n",
       "2             material entity             [bfo]   \n",
       "3                    electron  [chebi_ontology]   \n",
       "4      2-hydroxyglutarate(2-)  [chebi_ontology]   \n",
       "...                       ...               ...   \n",
       "26365           OPLAH (human)            [gene]   \n",
       "26366             PAH (human)            [gene]   \n",
       "26367             PRL (human)            [gene]   \n",
       "26368             PTH (human)            [gene]   \n",
       "26369             REN (human)            [gene]   \n",
       "\n",
       "                                                 parents  \n",
       "0                                  [b91618, BFO_0000020]  \n",
       "1                                          [BFO_0000040]  \n",
       "2                                          [BFO_0000004]  \n",
       "3                                          [CHEBI_36338]  \n",
       "4      [b67169, CHEBI_28965, CHEBI_18059, b67170, CHE...  \n",
       "...                                                  ...  \n",
       "26365                               [b64807, SO_0001217]  \n",
       "26366                               [b67084, SO_0001217]  \n",
       "26367                              [b133804, SO_0001217]  \n",
       "26368                              [b141039, SO_0001217]  \n",
       "26369                                [b3505, SO_0001217]  \n",
       "\n",
       "[26370 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26370\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 26370 entries, 0 to 26369\n",
      "Data columns (total 5 columns):\n",
      " #   Column       Non-Null Count  Dtype   \n",
      "---  ------       --------------  -----   \n",
      " 0   description  22753 non-null  object  \n",
      " 1   id           26370 non-null  object  \n",
      " 2   name         26370 non-null  object  \n",
      " 3   namespace    26370 non-null  category\n",
      " 4   parents      26363 non-null  object  \n",
      "dtypes: category(1), object(4)\n",
      "memory usage: 1.0+ MB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 47.9 s, sys: 1.8 s, total: 49.7 s\n",
      "Wall time: 48 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "bad_datas = []\n",
    "for i in dirs:\n",
    "    print(i.strip(\"./\"))\n",
    "    df = pd.read_parquet(i) \n",
    "    ## use with replace? slow! ,na_values=[\"None\",[],\"[]\"] \n",
    "    df.drop(columns=[\"obsoleteTerms\",\n",
    "                     \"dbXRefs\", # ? keep or not? e.g. for hpo \n",
    "                     \"crossReferences\",\"urls\",\"references\",\n",
    "                     \"obsoleteNames\",\"obsolete_terms\",\n",
    "                     \"obsoleteSymbols\",\n",
    "#                      \"textMiningSentences\",\"country\",\n",
    "                     \"betaConfidenceIntervalLower\",\"betaConfidenceIntervalUpper\",\n",
    "                     \"textMiningSentences\", ## causes error with array shape, maybe large?\n",
    "#                      \"chemblIds\"\n",
    "                     \"tep.url\",\"dir0\",\"indirectLocationIds\",\n",
    "                     \"ontology.leaf\",\n",
    "                     \"ontology.sources.url\"\n",
    "                    ],errors=\"ignore\",inplace=True)   \n",
    "    display(df) \n",
    "    ### SB can't handle list types - convert to categorical or string.. \n",
    "    total_shape = df.shape[0]\n",
    "    print(total_shape)\n",
    "    for c in df.select_dtypes(\"O\").columns:\n",
    "        df[c] = df[c].astype(str).replace(\"None\",np.NaN)\n",
    "#         df = df.replace(\"None\",np.NaN)\n",
    "        if df[c].nunique()< (total_shape/10):\n",
    "            if total_shape>1234567:\n",
    "                print(c)\n",
    "            df[c] = df[c].astype(\"category\")\n",
    "\n",
    "    try: df.dropna(axis=1,how=\"all\",inplace=True)        \n",
    "    except: ()\n",
    "    try: df.drop_duplicates(inplace=True)\n",
    "    except: df = df.loc[df.astype(str).drop_duplicates().index] # drop dupes when there's lists\n",
    "    display(df.info(memory_usage=True))\n",
    "    \n",
    "    try:\n",
    "        df.to_parquet(\"./proc/\"+i.strip(\"./\").replace(r\"/\",\"_\")+\".parquet\",index=False)\n",
    "#     except:\n",
    "#         try:\n",
    "#             df.infer_objects().to_parquet(\"./proc/\"+i.strip(\"./\").replace(r\"/\",\"_\")+\".parquet\",index=False,\n",
    "#                                          engine=\"pyarrow\")\n",
    "    except:\n",
    "        print(\"Failed to save\",i.strip(\"./\"))\n",
    "        bad_datas.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e7f716de",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.to_parquet(\"./proc/\"+i.strip(\"./\").replace(r\"/\",\"_\")+\".parquet\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8a853dc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sourceDatabase</th>\n",
       "      <th>targetA</th>\n",
       "      <th>intA</th>\n",
       "      <th>intABiologicalRole</th>\n",
       "      <th>targetB</th>\n",
       "      <th>intB</th>\n",
       "      <th>intBBiologicalRole</th>\n",
       "      <th>speciesA</th>\n",
       "      <th>speciesB</th>\n",
       "      <th>count</th>\n",
       "      <th>scoring</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>intact</td>\n",
       "      <td>ENSG00000182667</td>\n",
       "      <td>Q9P121-3</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000106331</td>\n",
       "      <td>Q3KNR5</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>intact</td>\n",
       "      <td>ENSG00000116171</td>\n",
       "      <td>P22307-2</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000105176</td>\n",
       "      <td>O94763-2</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>intact</td>\n",
       "      <td>ENSG00000115138</td>\n",
       "      <td>P01189</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000150636</td>\n",
       "      <td>Q68D86</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>intact</td>\n",
       "      <td>ENSG00000146530</td>\n",
       "      <td>Q8N2E2</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000204548</td>\n",
       "      <td>Q5J5C9</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>intact</td>\n",
       "      <td>ENSG00000112640</td>\n",
       "      <td>Q14738-2</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q8N5D4</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12189708</th>\n",
       "      <td>string</td>\n",
       "      <td>ENSG00000215182</td>\n",
       "      <td>ENSP00000485659</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000173757</td>\n",
       "      <td>ENSP00000293328</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12189709</th>\n",
       "      <td>string</td>\n",
       "      <td>ENSG00000215182</td>\n",
       "      <td>ENSP00000485659</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000135838</td>\n",
       "      <td>ENSP00000258317</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12189710</th>\n",
       "      <td>string</td>\n",
       "      <td>ENSG00000215182</td>\n",
       "      <td>ENSP00000485659</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000205426</td>\n",
       "      <td>ENSP00000369349</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12189711</th>\n",
       "      <td>string</td>\n",
       "      <td>ENSG00000215182</td>\n",
       "      <td>ENSP00000485659</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000196611</td>\n",
       "      <td>ENSP00000322788</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12189712</th>\n",
       "      <td>string</td>\n",
       "      <td>ENSG00000100129</td>\n",
       "      <td>ENSP00000485663</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>ENSG00000105372</td>\n",
       "      <td>ENSP00000470972</td>\n",
       "      <td>unspecified role</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>{'mnemonic': 'human', 'scientific_name': 'Homo...</td>\n",
       "      <td>4</td>\n",
       "      <td>0.983</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12189713 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         sourceDatabase          targetA             intA intABiologicalRole  \\\n",
       "0                intact  ENSG00000182667         Q9P121-3   unspecified role   \n",
       "1                intact  ENSG00000116171         P22307-2   unspecified role   \n",
       "2                intact  ENSG00000115138           P01189   unspecified role   \n",
       "3                intact  ENSG00000146530           Q8N2E2   unspecified role   \n",
       "4                intact  ENSG00000112640         Q14738-2   unspecified role   \n",
       "...                 ...              ...              ...                ...   \n",
       "12189708         string  ENSG00000215182  ENSP00000485659   unspecified role   \n",
       "12189709         string  ENSG00000215182  ENSP00000485659   unspecified role   \n",
       "12189710         string  ENSG00000215182  ENSP00000485659   unspecified role   \n",
       "12189711         string  ENSG00000215182  ENSP00000485659   unspecified role   \n",
       "12189712         string  ENSG00000100129  ENSP00000485663   unspecified role   \n",
       "\n",
       "                  targetB             intB intBBiologicalRole  \\\n",
       "0         ENSG00000106331           Q3KNR5   unspecified role   \n",
       "1         ENSG00000105176         O94763-2   unspecified role   \n",
       "2         ENSG00000150636           Q68D86   unspecified role   \n",
       "3         ENSG00000204548           Q5J5C9   unspecified role   \n",
       "4                     NaN           Q8N5D4   unspecified role   \n",
       "...                   ...              ...                ...   \n",
       "12189708  ENSG00000173757  ENSP00000293328   unspecified role   \n",
       "12189709  ENSG00000135838  ENSP00000258317   unspecified role   \n",
       "12189710  ENSG00000205426  ENSP00000369349   unspecified role   \n",
       "12189711  ENSG00000196611  ENSP00000322788   unspecified role   \n",
       "12189712  ENSG00000105372  ENSP00000470972   unspecified role   \n",
       "\n",
       "                                                   speciesA  \\\n",
       "0         {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "1         {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "2         {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "3         {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "4         {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "...                                                     ...   \n",
       "12189708  {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "12189709  {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "12189710  {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "12189711  {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "12189712  {'mnemonic': 'human', 'scientific_name': 'Homo...   \n",
       "\n",
       "                                                   speciesB  count  scoring  \n",
       "0         {'mnemonic': 'human', 'scientific_name': 'Homo...      3    0.560  \n",
       "1         {'mnemonic': 'human', 'scientific_name': 'Homo...      1    0.350  \n",
       "2         {'mnemonic': 'human', 'scientific_name': 'Homo...      3    0.560  \n",
       "3         {'mnemonic': 'human', 'scientific_name': 'Homo...      1    0.350  \n",
       "4         {'mnemonic': 'human', 'scientific_name': 'Homo...      1    0.350  \n",
       "...                                                     ...    ...      ...  \n",
       "12189708  {'mnemonic': 'human', 'scientific_name': 'Homo...      1    0.177  \n",
       "12189709  {'mnemonic': 'human', 'scientific_name': 'Homo...      2    0.206  \n",
       "12189710  {'mnemonic': 'human', 'scientific_name': 'Homo...      2    0.212  \n",
       "12189711  {'mnemonic': 'human', 'scientific_name': 'Homo...      2    0.229  \n",
       "12189712  {'mnemonic': 'human', 'scientific_name': 'Homo...      4    0.983  \n",
       "\n",
       "[12189713 rows x 11 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "eb50fe13",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.to_parquet(\"./proc/\"+i.strip(\"./\").replace(r\"/\",\"_\")+\".parquet\",index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c33d984d",
   "metadata": {},
   "source": [
    "Some dfs that fail:\n",
    "* evidence\n",
    "    * fastparquet OverflowError: value too large to convert to int\n",
    "    * try :` engine='pyarrow'`\n",
    "* drugWarnings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f79718c8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "91642c36",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_mol = pd.read_parquet(\"./proc/molecule.parquet\")\n",
    "# display(df_mol.info(memory_usage=True))\n",
    "# display(df_mol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0a988949",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ### SB can't handle list types - convert to categorical or string.. \n",
    "# for c in df_mol.select_dtypes(\"O\").columns:\n",
    "#     df_mol[c] = df_mol[c].astype(str).astype(\"category\")\n",
    "# df_mol.to_parquet(\"./proc/molecule_cat.parquet\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "597545ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_hpo = pd.read_parquet(\"./proc/hpo.parquet\")\n",
    "# display(df_hpo.info(memory_usage=True))\n",
    "# display(df_hpo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "926b8e71",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
